neqo-transport/src/connection/mod.rs

```diff
@@ -2880,8 +2885,8 @@ impl Connection {
     /// to check the estimated max datagram size and to use smaller datagrams.
     /// `max_datagram_size` is just a current estimate and will change over
     /// time depending on the encoded size of the packet number, ack frames, etc.
-    pub fn add_datagram(&mut self, buf: &[u8]) -> Res<bool> {
-        self.quic_datagrams.add_datagram(buf)
+    pub fn add_datagram(&mut self, buf: &[u8], id: Option<u64>) -> Res<()> {
```

Comment:
I wonder if we can make a new type for this, just to improve the readability of call sites:

```rust
enum DatagramTracking {
  None,
  Id(u64),
}
```

We can implement `Into<Option<u64>>` and `From<Option<u64>>` for people and even make this signature something like this so that `Some(20)` works (and your tests don't need to change again...):

```suggestion
    pub fn add_datagram(&mut self, buf: &[u8], id: impl Into<DatagramTracking>) -> Res<()> {
```

(Unrelated thought...) we have send_stream(), so why not just send_datagram() ?  "adding" a datagram is more how the operation works that what the caller wants to have happen.  They want to send a datagram.  That we have to add it to a queue is our business.

Reply:
This is a good suggestion. I will make this change.